main.c

```diff
@@ -364,6 +364,40 @@ static int socket_bindlisten(const char *socket_path,
   return -1;
 }
 
+static void remove_pidfile(const char *pidfile)
+{
+  if (unlink(pidfile) != 0) {
+    ERRORF("Failed to remove pidfile: \"%s\": %s", pidfile, strerror(errno));
+  }
+}
+
+static int create_pidfile(const char *pidfile)
+{
+  int flags = O_WRONLY | O_CREAT | O_EXLOCK | O_TRUNC | O_NONBLOCK;
+  int pid_fd = open(pidfile, flags, 0644);
```

Comment:
nit:
```suggestion
  int fd = open(pidfile, flags, 0644);
```

To me, `pid_fd` sounds like Linux's pidfd https://man7.org/linux/man-pages/man2/pidfd_open.2.html

Reply:
Yes, I also thought about pidfd. We also rename it to pidfile_fd in main to avoid the this confusion.

Reply:
Renamed